update: Reflect the current state of MV2 support for uBlock Origin (#3250)
privacyguides/privacyguides.org: docs/browser-extensions.md

[:octicons-info-16:](https://github.com/gorhill/uBlock/wiki){ .card-link title=Documentation}
[:octicons-code-16:](https://github.com/gorhill/uBlock){ .card-link title="Source Code" }

<details class="admonition info" markdown>
<summary>Support</summary>

uBlock Origin being is only supported on Firefox-based browsers and a handful of Chromium browsers (Brave and Opera).

</details>

> 

Reviewer: ```suggestion
```
We don't need to warn people about what platforms are supported, you can just see in the downloads section or on their website.

Author: You can't really though. It is still listed on Chrome Web Store despite not being available on Chrome (it is still relevant to list the CWS being Google allows other Chromium browser users to download from its store)

Author: Also, you can download uBlock Origin in Brave Settings for when Google will remove it from its store.

Author: Finally, I feel this need to be said upfront since we have uBlock Origin in front even though it's not available on most browsers.

<details class="downloads" markdown>
<summary>Downloads</summary>

> - [:simple-firefoxbrowser: Firefox](https://addons.mozilla.org/firefox/addon/ublock-origin)
> - [:simple-googlechrome: Chrome](https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm)

Reviewer: ```suggestion
- [:simple-firefoxbrowser: Firefox](https://addons.mozilla.org/firefox/addon/ublock-origin)
- [:fontawesome-brands-edge: Edge](https://microsoftedge.microsoft.com/addons/detail/ublock-origin/odfafepnkmbhccpbejgmiehpchacaeak)
```

Reviewer: The Chrome version doesn't work anymore but the Edge version still works.

Author: See my comment above. Source that Edge allows other browsers to download from its site or that Edge still supports MV2 ?

Reviewer: > See my comment above. Source that Edge allows other browsers to download from its site or that Edge still supports MV2 ?

I mean I literally just did it to test it out. They still offer it on their [store](https://microsoftedge.microsoft.com/addons/detail/ublock-origin/odfafepnkmbhccpbejgmiehpchacaeak) and it still works. It's also officially linked by the uBlock Origin GitHub.

Well they don't let other browsers install from their store but they do let you install it in Edge and it works fine is what I mean.

Author: We will need to check on Windows. It indeed still works on Linux but in any case, Microsoft has **never** made any promise to keep support, so we shouldn't recommend people to download something which might be suddenly disabled next week.

Reviewer: > We will need to check on Windows. It indeed still works on Linux but in any case, Microsoft has **never** made any promise to keep support, so we shouldn't recommend people to download something which might be suddenly disabled next week.

I checked on Windows, it works fine. When that happens we can update the site.

Author: With the dev version based on Chrome 151 ?

Reviewer: > With the dev version based on Chrome 151 ?

I had a fresh install of Windows 11 in a VM and updated everything and it still works.

<summary>Downloads</summary>

> - [:simple-firefoxbrowser: Firefox](https://addons.mozilla.org/firefox/addon/ublock-origin)

Reviewer: ```suggestion
- [:simple-firefoxbrowser: Firefox](https://addons.mozilla.org/firefox/addon/ublock-origin)
- [:simple-googlechrome: Chrome](https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm)
```

- ...you don't want to grant full "read/modify website data" permissions to any extensions (even a trusted one like uBlock Origin)
- ...you want a more resource (memory/CPU) efficient content blocker[^1]

> - ...your browser only supports Manifest V3 extensions. This is the case for Chrome, Edge and most Chromium browsers.[^2]

Reviewer: ```suggestion
- ...your browser only supports Manifest V3 extensions. This is the case for Chrome[^2].
```

Author: Again, as above, there is absolutely no proof Edge will keep MV2 support.

Reviewer: > Again, as above, there is absolutely no proof Edge will keep MV2 support.

Yeah but they still do for now so we're reflecting the current situation. It can be updated later when they drop support.

Author: Fair enough, but we do need to warn that Chromium is removing support.

Author: Also, please don't resolve conversation when they aren't resolved this feels quite annoying.

Reviewer: > Also, please don't resolve conversation when they aren't resolved this feels quite annoying.

It resolves when I merge the changes but then when you reply to me it unresolves it seemingly.

[^1]: uBlock Origin Lite *itself* will consume no resources, because it uses newer APIs which make the browser process the filter lists natively, instead of running JavaScript code within the extension to handle the filtering. However, this resource advantage is only [theoretical](https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)#is-ubol-more-efficient-cpu--and-memory-wise-than-ubo), because it's possible that standard uBlock Origin's filtering code is more efficient than your browser's native filtering code. This has not yet been benchmarked.

> [^2]: As of June 2026, [Brave](https://brave.com/blog/brave-shields-manifest-v3/) and [Opera](https://xcancel.com/Opera_Security/status/2066543496001888753) have committed to maintain MV2 support for the foreseeable future. Be aware that any Chromium-based browser that still supports MV2 extensions is now responsible for maintaining and securing the infrastructure as Google progressively removes MV2 support from the codebase.

Reviewer: ```suggestion
[^2]: [Brave](https://brave.com/blog/brave-shields-manifest-v3/) and [Opera](https://xcancel.com/Opera_Security/status/2066543496001888753) have committed to maintain MV2 support for as long as they're able to.
```

[^1]: uBlock Origin Lite *itself* will consume no resources, because it uses newer APIs which make the browser process the filter lists natively, instead of running JavaScript code within the extension to handle the filtering. However, this resource advantage is only [theoretical](https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)#is-ubol-more-efficient-cpu--and-memory-wise-than-ubo), because it's possible that standard uBlock Origin's filtering code is more efficient than your browser's native filtering code. This has not yet been benchmarked.

> [^2]: As of June 2026, [Brave](https://brave.com/blog/brave-shields-manifest-v3/) and [Opera](https://xcancel.com/Opera_Security/status/2066543496001888753) have commited to maintain MV2 support for the foreseable future. Be aware that any chromium-based browser that still supports MV2 extensions is now responsible for maintaining and securing the infrastructure as Google progressively removes MV2 support from the codebase.

Reviewer: ```suggestion
[^2]: As of June 2026, [Brave](https://brave.com/blog/brave-shields-manifest-v3/) and [Opera](https://xcancel.com/Opera_Security/status/2066543496001888753) have committed to maintain MV2 support for the foreseeable future. Be aware that any Chromium-based browser that still supports MV2 extensions is now responsible for maintaining and securing the infrastructure as Google progressively removes MV2 support from the codebase.
```

Author: I don't get your position. Opera said in september 2025 - and stated this month that the blog post was still correct - that they "are **actively working to keep** it that way [**MV2 support**] for **as long as it’s technically reasonable.**"

And **Brave** very explicitly said that  they will "offer l**imited MV2 support even after it’s fully removed from the upstream Chromium codebase**."

Sources are in the PR, also check this (and comment at the end) article from [NeoWin](https://www.neowin.net/news/google-chrome-is-killing-all-ublock-origin-bypasses-microsoft-edge-opera-to-follow/)
